pkg/controller/utils/pod/pod.go

```diff
@@ -130,22 +130,37 @@ func MostRecentRestart(pod *v1.Pod) (time.Time, datadoghqv1alpha1.ExtendedDaemon
 		if s.RestartCount != 0 && s.LastTerminationState != (v1.ContainerState{}) && s.LastTerminationState.Terminated != (&v1.ContainerStateTerminated{}) {
 			if s.LastTerminationState.Terminated.FinishedAt.After(restartTime) {
 				restartTime = s.LastTerminationState.Terminated.FinishedAt.Time
-				reason = terminatedReason(s.LastTerminationState.Terminated.Reason)
+				reason = datadoghqv1alpha1.ExtendedDaemonSetStatusReason(s.LastTerminationState.Terminated.Reason)
 			}
 		}
 	}
 	return restartTime, reason
 }
 
-func terminatedReason(reason string) datadoghqv1alpha1.ExtendedDaemonSetStatusReason {
-	switch reason {
-	case string(datadoghqv1alpha1.ExtendedDaemonSetStatusReasonCLB):
-		return datadoghqv1alpha1.ExtendedDaemonSetStatusReasonCLB
-	case string(datadoghqv1alpha1.ExtendedDaemonSetStatusReasonOOM):
-		return datadoghqv1alpha1.ExtendedDaemonSetStatusReasonOOM
-	default:
-		return datadoghqv1alpha1.ExtendedDaemonSetStatusReasonUnknown
+var cannotStartReasons = []string{
+	"ErrImagePull",
+	"ImagePullBackOff",
+	"CreateContainerConfigError",
+}
```

Comment:
could you add other reasons defined here: https://github.com/kubernetes/kubernetes/blob/88512be213159f87ed2dad07f75c02d355eadac5/pkg/kubelet/kuberuntime/kuberuntime_container.go#L58